### PR TITLE
Remove legacy renumbering

### LINKED ISSUE #1679: [BUG] Intermittent Hangs on MNMG
There is an intermittent hang on Prom which goes away after rerunning. Here is a [link](https://gist.github.com/ayushdg/e2fb1cd8611bf5b17769f42bf2fbfa18) to the error which can be located in the log file. There is no clear way to reproduce the error if not running the benchmarks couple times. That error can be found in combination with an OOM (maximum pool size exceeded) if you run BFS (unweighted dataset) with 6 Nodes at scale 30.

### LINKED ISSUE #2437: [BUG] Fix exposed `legacy_renum_only` options in public graph APIs
A change (still to be merged at the time of this writing) exposes a `legacy_renum_only` which was previously an internal flag, but is now exposed in order to create graphs specifically for use with algos that use pylibcugraph Graph objects stored in the cuGraph graph objects.

This is confusing to users not familiar with the implementation details, and could cause problems if users don't understand when they should or shouldn't set it.

For reference: https://github.com/rapidsai/cugraph/pull/2394#discussion_r927139725

<hr>

Behavior based on `legacy_renum_only` setting vs. SG/MG/algo type

<table>
<tr><td></td><td><b>SG legacy algo</b></td><td><b>MG legacy algo</b></td><td><b>SG plc algo</b></td><td><b>MG plc algo</b></td></tr>
<tr><td><b>legacy_renum_only=True</b></td><td><b>ERROR</b></td><td><b>ERROR</b></td><td>optimal performance</td><td>optimal performance</td></tr>
<tr><td><b>legacy_renum_only=False</b></td><td>no error</td><td>no error</td><td>no error but reduced performance (redundant renumber step)</td><td>no error but reduced performance (redundant renumber step), and <b>potential hang</b></td></tr>
</table>

### LINKED ISSUE #2535: MNMG Induce Subgraph - Python
Modify the python induce subgraph calls to use the new PLC version of induce subgraph. Add MNMG support.

API definition in https://github.com/rapidsai/cugraph/issues/2529 blocks starting this work
Implementation in https://github.com/rapidsai/cugraph/issues/2532 blocks completing this work

### LINKED ISSUE #2536: MNMG Induce Subgraph - Python unit tests
Implement Induce Subgraph unit tests for python.

### LINKED ISSUE #2537: MNMG Induce Subgraph - Remove old Python
Remove old python implementation of induce subgraph.

### PR DESCRIPTION
**Pull request by jnke2016**
closes #2437 
closes #1679
closes #2535 
closes #2536 
closes #2537

### CODE REVIEW

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
rerun tests

File: python/cugraph/cugraph/tests/conftest.py [hunk 1]
```diff
@@ -36,11 +36,11 @@ def dask_client():
         enable_tcp_over_ucx=True,
         enable_nvlink=True,
         enable_infiniband=True,
-        enable_rdmacm=True,
+        enable_rdmacm=False,
         # net_devices="mlx5_0:1",
```
Comment (VibhuJawa) [inline python/cugraph/cugraph/tests/conftest.py:40]: I suggest setting everything to `None`:
I think this not aligned with the current conftest

https://github.com/rapidsai/cugraph/blob/6c89b6359f853d9558d72c47d98e16615e3b329a/python/cugraph/cugraph/testing/mg_utils.py#L26-L28


You will have to make align it here.

File: python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py [hunk 2]
```diff
@@ -195,9 +201,7 @@ def __from_edgelist(
             )
             source = renumber_map.renumbered_src_col_name
             destination = renumber_map.renumbered_dst_col_name
-            # Use renumber_map to figure out if renumbering was skipped or not
-            # This was added to handle 'legacy_renum_only' which will skip the
-            # old C++ renumbering  when running the pylibcugraph/C algos
+            # Use renumber_map to figure out if the python renumbering occured
             self.properties.renumbered = renumber_map.implementation.numbered
```
Comment (VibhuJawa) [inline python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py:205]: I seem to be getting below error  on this, can you recheck

```
stributedGraphImpl.__from_edgelist(self, input_ddf, source, destination, edge_attr, renumber, store_transposed, legacy_renum_only)
    232 # If renumbering is not enabled, this function will only create
    233 # the edgelist_df and not do any renumbering.
    234 # C++ renumbering is enabled by default for algorithms that
    235 # support it (but only called if renumbering is on)
    236 self.compute_renumber_edge_list(
    237     transposed=store_transposed, legacy_renum_only=legacy_renum_only
    238 )
--> 240 self.properties.renumbered = self.renumber_map.implementation.numbered
    241 ddf = self.edgelist.edgelist_df
    243 num_edges = len(ddf)

AttributeError: 'NoneType' object has no attribute 'implementation'
```

File: python/cugraph/cugraph/structure/number_map.py [hunk 3]
```diff
@@ -550,8 +516,8 @@ def renumber_and_segment(
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/structure/number_map.py:516]: Can we get a comment here explaining that renumbering only occurs here if the column names are a list?

File: python/cugraph/cugraph/structure/number_map.py [hunk 4]
```diff
@@ -481,7 +454,13 @@ def renumber_and_segment(
         store_transposed=False,
         legacy_renum_only=False,
     ):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/structure/number_map.py:456]: I think it would be a good time to add a docstring for this function that explains exactly what it does and why.

File: python/cugraph/cugraph/tests/mg/test_mg_graph.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2022, NVIDIA CORPORATION.
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/mg/test_mg_graph.py:1]: Can you add an additional test to this file (and sg as well) to ensure renumbering works as expected when edge ids, edge types, and weights are passed in as the edge attributes?

**Comment by alexbarghi-nv** [comment]
Shouldn't the renumber wrapper get thrown out in this PR?
(python/cugraph/cugraph/structure/renumber_wrapper.pyx)

**Comment by alexbarghi-nv** [comment]
This PR appears to break personalized pagerank tests.

File: python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py (hunk 2 above)
Comment (rlratzel) [reply python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py:205]: @jnke2016 it might be better to add a proper method or bool property to NumberMap, like `is_renumbered` instead of trying to access the internal objects.

File: python/cugraph/cugraph/structure/graph_classes.py [hunk 6]
```diff
@@ -282,13 +282,8 @@ def from_dask_cudf_edgelist(
             If True, stores the transpose of the adjacency matrix.  Required
             for certain algorithms.
 
-        legacy_renum_only : bool, optional (default=False)
```
Comment (rlratzel) [inline python/cugraph/cugraph/structure/graph_classes.py:285]: You probably shouldn't remove the description if the arg is still available to be used.  You should instead mention that it's deprecated and/or explain how it's supposed to be used.  If it's being deprecated here, then we should also be raising a warning.

Also, shouldn't we be updating `from_cudf_edgelist()` too?

**Review by rlratzel — CHANGES_REQUESTED** [review]

File: python/cugraph/cugraph/tests/conftest.py [hunk 7]
```diff
@@ -36,11 +36,11 @@ def dask_client():
...
     )
 
     yield client
 
-    stop_dask_client(client)
+    stop_dask_client(*client)
```
Comment (eriknw) [inline python/cugraph/cugraph/tests/conftest.py:45]: Why `*client` here? Perhaps `stop_dask_client(client, cluster)` instead?

**Comment by seunghwak** [comment]
Are we using shuffle.py (https://github.com/rapidsai/cugraph/blob/branch-23.04/python/cugraph/cugraph/structure/shuffle.py) in anyplace? Can't we delete this file if no longer relevant?

**Review by ChuckHastings — APPROVED** [review]

File: python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py (hunk 2 above)
Comment (jnke2016) [reply python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py]: I believe I resolved this issue per @rlratzel recommendation but feel free to test it again and let me know @VibhuJawa

File: python/cugraph/cugraph/tests/conftest.py (hunk 1 above)
Comment (jnke2016) [reply python/cugraph/cugraph/tests/conftest.py]: I believe this was resolved when I fetched the latest changes

File: python/cugraph/cugraph/structure/number_map.py (hunk 3 above)
Comment (jnke2016) [reply python/cugraph/cugraph/structure/number_map.py:525]: Sure. I added it.

File: python/cugraph/cugraph/structure/number_map.py (hunk 4 above)
Comment (jnke2016) [reply python/cugraph/cugraph/structure/number_map.py:466]: Done.

File: python/cugraph/cugraph/structure/graph_classes.py (hunk 6 above)
Comment (jnke2016) [reply python/cugraph/cugraph/structure/graph_classes.py:292]: Right. I believe this was resolved.

File: python/cugraph/cugraph/tests/conftest.py (hunk 7 above)
Comment (jnke2016) [reply python/cugraph/cugraph/tests/conftest.py]: Yep. resolved in a previous PR

File: python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py (hunk 2 above)
Comment (VibhuJawa) [reply python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py]: I think your tests should catch if the error would have happened.

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Comment by jnke2016** [comment]
rerun tests

File: python/cugraph/cugraph/community/subgraph_extraction.py [hunk 8]
```diff
@@ -56,33 +57,12 @@ def subgraph(G, vertices):
 
     """
 
-    G, isNx = ensure_cugraph_obj_for_nx(G)
-    directed = G.is_directed()
+    warning_msg = (
+        "This call is deprecated and will be refactored "
+        "in the next release. Please call 'cugraph.induced_subgraph()' instead"
+    )
+    warnings.warn(warning_msg, PendingDeprecationWarning)
```
Comment (rlratzel) [inline python/cugraph/cugraph/community/subgraph_extraction.py:64]: Should use `DeprecationWarning` since it's now being officially deprecated.
```suggestion
    warnings.warn(warning_msg, DeprecationWarning)
```

File: python/cugraph/cugraph/structure/graph_classes.py [hunk 9]
```diff
@@ -153,6 +153,8 @@ def from_cudf_edgelist(
             pylibcugraph algorithms.  Must be false for algorithms
             not yet converted to the pylibcugraph C API.
 
+            This parameter is deprecated and will be removed.
```
Comment (rlratzel) [inline python/cugraph/cugraph/structure/graph_classes.py:156]: Can you also raise a `DeprecationWarning` if the user sets `legacy_renum_only` to `True`?

**Review by rlratzel — CHANGES_REQUESTED** [review]

**Review by VibhuJawa — APPROVED** [review]
LGTM

File: python/cugraph/cugraph/structure/graph_classes.py (hunk 9 above)
Comment (jnke2016) [reply python/cugraph/cugraph/structure/graph_classes.py:156]: So this function eventually calls `self._Impl._simpleGraphImpl__from_edgelist` which intrinsically raises the `DeprecationWarning`. I didn't add an extra one to avoid duplicates

**Review by rlratzel — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]
